src/pages/signin/LoginForm.js

```diff
@@ -36,26 +40,39 @@ const propTypes = {
 };
 
 const defaultProps = {
-    account: {},
+    account: {
+        login: '',
+    },
 };
 
 class LoginForm extends React.Component {
     constructor(props) {
         super(props);
 
+        this.updateLogin = this.updateLogin.bind(this);
         this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
+        this.login = props.account.login;
 
         this.state = {
             formError: false,
-            login: '',
         };
     }
 
+    /**
+     * Update the value of login in Onyx
+     *
+     * @param {String} newLogin
+     */
+    updateLogin(newLogin) {
+        this.login = newLogin;
```

Comment:
Not sure I get why we are setting `this.login = newLogin`. Why not just use the value we are storing in Onyx?

Reply:
https://github.com/Expensify/App/pull/4385#issuecomment-892160103
> We just won't use that value directly in the form element since that will have bad performance.

Don't know if I understood this properly.

Reply:
I'll explain some more. I don't think we need to update this value anywhere besides the Onyx key. The `defaultValue` can use `this.props.account.login` since the only time we need it is when the component mounts again?

Reply:
Got it. 
Just wanted to clarify. If we're going to use `Onyx`for `defaultValue` then there's no use of `ref=` now, right?

Reply:
I don't think we ever need to use the ref. It's an incorrect usage of ref that is just getting overwritten by the `updateLogin()` method.